### DISCUSSION TITLE
Frontend forms with just Hono?

### DISCUSSION THREAD

**Discussion by osseonews** [honojs/hono#2405] [Q&A]
I use Hono on Workers, to interface with a Nextjs frontend. However, I just noticed that Hono is a framework on Cloudflare Pages.  I'm wondering then if it's possible to just create a simple frontend website with Hono on Cloudflare pages which will show a user a form (or forms) to fill out. I was looking at the html helper in Hono, but I'm not clear how to use that to set up the form. 

BTW, the use case for this, is that I'm creating an AI application and for nearly all basic apps, all you need is a form prompt, which is then sent to Cloudflare Workers AI. To set up an entire Nextjs website to just have a basic form input is obviously overkill. If we can just use Hono (which we use already) to generate the frontend form also, then we can have a full AI application running on hono on cloudflare pages quite easily. I'm guessing someone has done this already? If so, is there any sample code?

Thanks very much.

**Comment by emmanuelchucks** [comment]
Look at this: https://github.com/yusukebe/my-first-workers-ai

**Reply by osseonews** [reply]
Wow, thank you! That code is brilliant. Easiest implementation of AI I have seen yet. I'm wondering, how you could in theory, save the each form query, so the UI could appear as a chat? Also, I'm curious, as Hono has a JSX renderer, why couldn't the approach in that repo be used to create a basic website that interactive, and you would have no need for massive frameworks like Nextjs etc.

**Reply by emmanuelchucks** [reply]
In theory, prepend the array of previous messages minus the first 'system' message as `p` tags before the current streaming response. 
https://github.com/yusukebe/my-first-workers-ai/blob/a6a7eb9583d966b1ab8359dee7cccb4443bdc05c/assets/script.js#L1-L6

Hono is still growing, not as battle-tested as Next.js. That alone might scare some.
